### PR TITLE
Added the automatically generated documentation

### PR DESCRIPTION
**Pull request by Mailaender**
due to https://github.com/OpenRA/OpenRA/issues/14225

### CODE REVIEW

File: src/modding/documentation/lua.md [hunk 1]
```diff
@@ -0,0 +1,959 @@
+This is an automatically generated listing of the new Lua map scripting API, generated for {DEV_VERSION} of OpenRA.
+
+OpenRA allows custom maps and missions to be scripted using Lua 5.1.
+These scripts run in a sandbox that prevents access to unsafe functions (e.g. OS or file access), and limits the memory and CPU usage of the scripts.
+
+You can access this interface by adding the [LuaScript](Traits#luascript) trait to the world actor in your map rules (note, you must replace the spaces in the snippet below with a single tab for each level of indentation):
```
Comment (abcdefg30) [inline src/modding/documentation/lua.md]: This link probably won't work?

Comment (Mailaender) [reply src/modding/documentation/lua.md]: Yes, it fails. But as this is auto-generated, it should be fixed elsewhere I guess.

Comment (Mailaender) [reply src/modding/documentation/lua.md]: Fixed it manually.

**Comment by abcdefg30** [comment]
Ping @pchote - you wanted to comment something here before I merge this.

**Comment by pchote** [comment]
I had two issues which were raised and partly discussed in IRC:
* IMO it doesn't make sense to categorize these pages under "Documentation". These pages are the least document-ey things in the book!
* The current layout conflicts with https://github.com/OpenRA/OpenRA/blob/bleed/packaging/update-wiki.sh - we need to support playtest/release versions.

I suggested that both issues can be fixed by structuring this as:

```markdown
    - [Reference API (Release)](./api/release/README.md)
        - [Traits](./api/release/traits.md)
        - [Lua API](./api/release/lua.md)
        - [Weapons](./api/release/weapons.md)
    - [Reference API (Playtest)](./api/playtest/README.md)
        - [Traits](./api/playtest/traits.md)
        - [Lua API](./api/playtest/lua.md)
        - [Weapons](./api/playtest/weapons.md)
```

The `./api/*/README.md` pages can then be similar to the current `./modding/documentation/README.md`, but have a title like "Reference API (release-20191117)", and the body can contain a link back to the github release for the tag. The update-wiki script can then regenerate the page content from a template hardcoded in its script.

**Comment by Mailaender** [comment]
I will try to revive https://openra.readthedocs.io/ instead. See https://github.com/OpenRA/docs

**Pull request closed without merging**